krm-functions/lib/nad/v1/nad.go

```diff
@@ -0,0 +1,364 @@
+	Addresses []Addresses `json:"addresses,omitempty"`
+}
+
+type Addresses struct {
+	Address string `json:"address,omitempty"`
+	Gateway string `json:"gateway,omitempty"`
+}
+
+type CniSpecType int64
+
+const (
+	BothIpamVlan CniSpecType = iota // 0
+	VlanAllocOnly
+	IpVlanType
+	SriovType
+	MacVlanType
+)
+
+type NadStruct struct {
+	K           kubeobject.KubeObjectExt[nadv1.NetworkAttachmentDefinition]
+	CniSpecType CniSpecType
+}
+
+// NewFromKubeObject creates a new parser interface
+// It expects a *fn.KubeObject as input representing the serialized yaml file
+func NewFromKubeObject(b *fn.KubeObject) (*NadStruct, error) {
+	p, err := kubeobject.NewFromKubeObject[nadv1.NetworkAttachmentDefinition](b)
+	if err != nil {
+		return nil, err
+	}
+	return &NadStruct{K: *p}, nil
+}
+
+// NewFromYAML creates a new parser interface
+// It expects a raw byte slice as input representing the serialized yaml file
+func NewFromYAML(b []byte) (*NadStruct, error) {
+	p, err := kubeobject.NewFromYaml[nadv1.NetworkAttachmentDefinition](b)
+	if err != nil {
+		return nil, err
+	}
+	return &NadStruct{K: *p}, nil
+}
+
+// NewFromGoStruct creates a new parser interface
+// It expects a go struct representing the interface krm resource
+func NewFromGoStruct(b *nadv1.NetworkAttachmentDefinition) (*NadStruct, error) {
+	p, err := kubeobject.NewFromGoStruct[nadv1.NetworkAttachmentDefinition](*b)
+	if err != nil {
+		return nil, err
+	}
+	return &NadStruct{K: *p}, nil
+}
+
+func (r *NadStruct) GetStringValue(fields ...string) string {
+	if r == nil {
+		return ""
+	}
+	s, ok, err := r.K.NestedString(fields...)
+	if err != nil {
+		return ""
+	}
+	if !ok {
+		return ""
+	}
+	return s
+}
+
+func (r *NadStruct) GetConfigSpec() string {
+	if r == nil {
+		return ""
+	}
+	return r.GetStringValue(ConfigType...)
+}
+
+func (r *NadStruct) GetExistingNadConfig() NadConfig {
+	nadConfigStruct := NadConfig{}
+	if err := json.Unmarshal([]byte(r.GetStringValue(ConfigType...)), &nadConfigStruct); err != nil {
+		return nadConfigStruct
+	}
+	return nadConfigStruct
+}
+
+func (r *NadStruct) GetCNIType() string {
+	for i, plugin := range r.GetExistingNadConfig().Plugins {
+		if plugin.Type == TuningType {
+			continue
+		} else {
+			return r.GetExistingNadConfig().Plugins[i].Type
+		}
+	}
+	return ""
+}
+
+func (r *NadStruct) GetVlan() int {
+	return r.GetExistingNadConfig().Vlan
+}
+
+func (r *NadStruct) GetNadMaster() string {
+	for i, plugin := range r.GetExistingNadConfig().Plugins {
+		if plugin.Type == TuningType {
+			continue
+		} else {
+			return r.GetExistingNadConfig().Plugins[i].Master
+		}
+	}
+	return ""
+}
+
+func (r *NadStruct) GetIpamAddress() []Addresses {
+	for i, plugin := range r.GetExistingNadConfig().Plugins {
+		if plugin.Type == TuningType {
+			continue
+		} else {
+			return r.GetExistingNadConfig().Plugins[i].Ipam.Addresses
+		}
+	}
+	return []Addresses{}
+}
+
+func (r *NadStruct) GetNewNadConfig() (NadConfig, error) {
```

Comment:
Suggest chaning the name to CreateNewNadConfig

Reply:
removed